jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageWriter.java

```diff
@@ -88,7 +88,13 @@ private static void moveIfDoesNotExist(Path source, Path destination) throws IOE
                     // previously-cached layer.
                     return true;
                   }
-                  Files.move(source, destination);
+                  try{
+                    Files.move(source, destination);
+                  } catch (FileSystemException e){
+                    String message = "Unable to copy file from %s to %s. This operation may not be supported by the" +
+                            "file system.";
+                    throw new FileSystemException(message);
```

Comment:
So a couple of things to keep in mind when doing this.

- Down below (retryOnException) is designed to retry this loop if a FileSystemException occurs. So we probably don't want to actually use this here.

- In intellij you can search for classes and inspect their contents (ctrl-shift-n ?). If we take a look at FileSystemException there a few constructors in there. The single string constructor actually expects a file as the first parameter. (We're probably not going to use FileSystemException here because of earlier point)

Reply:
Thanks for pointing that out! Moved this message down to the `if (!success))` condition.